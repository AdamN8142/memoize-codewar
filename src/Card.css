.solution-image {
  height: 500px;
  width: 750px;
  margin-left: 46px;
  margin-right: 20px;
  margin-bottom: 35px;
}

.individual-card {
  border:solid black;
  border-radius: 25px;
  margin-bottom: 60px;
  width: 850px;
  animation: slideInX 3.5s ease 0s 1 normal; 
  animation-fill-mode: forwards;
  background-image: url("https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSx_FVYfREyzYPsYobXZuXpXKIzu08n5owpMeno-LIjtyMcg8cG");
  background-size: cover;
  background-repeat: no-repeat;
  opacity: .9;
}


.card-name {
  font-size: 55px;
  margin-top: 29px;
  margin-bottom: 25px;
  text-align: center;
  /* font-family: 'Indie Flower', cursive; */
  font-family: 'Luckiest Guy', cursive;
  letter-spacing: .05em;

}

.card-prompt {
  font-size: 25px;
  font-family: 'Indie Flower', cursive;
  border:solid black 2px;
  border-radius:25px;
  margin-left: 35px;
  margin-right: 35px;
  padding: 25px;
  margin-bottom: 50px;
}

.level-and-category {
  display: flex;
  justify-content: center;
}

.card-category {
  margin-right: 35px;
  font-size: 45px;
  font-family: 'Indie Flower', cursive;
}

.card-level{
  margin-left: 35px;
  font-size: 45px;
  margin-bottom:20px;
  font-family: 'Indie Flower', cursive;

}

.repl-link {
  text-align: center;
  margin-bottom: 30px;
}

.link {
  display: flex;
  justify-content: center;
}


.card-buttons-container {
  display: flex;
  justify-content: space-evenly;
  margin: 0 auto;
  margin-bottom: 25px;
}


.save-to-storage-btn {
  height: 100px;
  width: 180px;
  border-radius: 50px;
  background-image:url("https://wallpapercave.com/wp/gfwPusm.jpg");
  color: white;
  border:solid 4px white;
  font-size: 18px;
}

.see-solution-btn {
  height: 100px;
  width: 180px;
  border-radius: 50px;
  background-image:url("https://wallpapercave.com/wp/gfwPusm.jpg");
  color:white;
  border:solid 4px white;
  font-size: 18px;
}

.see-solution-btn:hover {
  background-image: linear-gradient(purple, red, pink);
  color: black;
  opacity: .7;
}

.save-to-storage-btn:hover {
  background-image: linear-gradient(purple, red, pink);
  color: black;
  opacity: .7;
}

/* .card-button {
  height: 100px;
  width: 180px;
  border-radius: 50px;
  background-image:url("https://wallpapercave.com/wp/gfwPusm.jpg");
  color:white;
  border:solid 4px white;
  font-size: 18px;
} */