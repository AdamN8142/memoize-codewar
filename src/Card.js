import React, { Component } from 'react';
import CardContainer from './CardContainer';

class Card extends Component {
  constructor() {
    super();
  }

  toggleCompletedProperty = (event) => {
    this.props.toggleComplete(this.props.codeWar.id)
  }


  render() {
    return (
      <div>
        <article className="individual-card">
        <h3 className="card-name">{this.props.codeWar.name}</h3>
          <div className="level-and-category">
            <p className="card-category">Category : {this.props.codeWar.type}</p>
            <p className="card-level">Level : {this.props.codeWar.level}</p>
          </div>
          <div className="link">
            <a href={this.props.codeWar.replLink} target="_blank" className="repl-link">{this.props.codeWar.name} - REPL Link </a>
          </div>
          <h4 className="card-prompt">{this.props.codeWar.prompt}</h4>
          <div className="card-buttons-container">
            <button className="card-button" className="see-solution-btn"onClick={this.toggleCard}>See the Solution</button>
            {!this.props.codeWar.addedToStorage && <button className="card-button" className="save-to-storage-btn" onClick={this.toggleCompletedProperty}>Mark as Completed!</button>}
            {this.props.codeWar.addedToStorage && <button className="card-button" className="save-to-storage-btn" onClick={this.toggleCompletedProperty}>Remove from Completed!</button>}

          </div>
        </article>
      </div>
    )
  }
}

export default Card;