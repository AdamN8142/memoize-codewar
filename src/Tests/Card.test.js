import React from "react";
import Card from "../Card";
import { shallow } from "enzyme";
import { isFulfilled } from "q";

const mockProps = {
  toggleComplete: jest.fn(),
  key: 1,
  codeWar: {
    id: 1,
    name: "Abbreviate a Two Word Name",
    type: "String",
    prompt: "Write a function to convert a name into initials. This kata strictly takes two words with one space in between them.",
    level: "8 kyu",
    replLink: "https://repl.it/@codesForDays/Abbreviate-a-Two-Word-Name?language=nodejs&folderId=",
    addedToStorage: false,
    },
  completed: [{
    id: 1,
    name: "Abbreviate a Two Word Name",
    type: "String",
    prompt: "Write a function to convert a name into initials. This kata strictly takes two words with one space in between them.",
    level: "8 kyu",
    replLink: "https://repl.it/@codesForDays/Abbreviate-a-Two-Word-Name?language=nodejs&folderId=",
    addedToStorage: false,
    },]
}

describe('Card', () => {
  let wrapper; 

  beforeEach(()=> {
    wrapper = shallow (
      <Card 
      {...mockProps}/>
    )
  })

  it("should match the snapshot will all data passed in", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("should toggle the solution", () => {
    expect(wrapper.state()).toEqual({toggledSolution: false});
    wrapper.find(".see-solution-btn").first().simulate("click", { target: { id: 1}})
    expect(wrapper.state()).toEqual({toggledSolution: true})
  })

})