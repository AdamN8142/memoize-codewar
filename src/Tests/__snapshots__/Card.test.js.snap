// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot will all data passed in 1`] = `
<div>
  <article
    className="individual-card"
  >
    <h3
      className="card-name"
    >
      Abbreviate a Two Word Name
    </h3>
    <div
      className="level-and-category"
    >
      <p
        className="card-category"
      >
        Category : 
        String
      </p>
      <p
        className="card-level"
      >
        Level : 
        8 kyu
      </p>
    </div>
    <div
      className="link"
    >
      <a
        className="repl-link"
        href="https://repl.it/@codesForDays/Abbreviate-a-Two-Word-Name?language=nodejs&folderId="
        target="_blank"
      >
        Abbreviate a Two Word Name
         - REPL Link 
      </a>
    </div>
    <h4
      className="card-prompt"
    >
      Write a function to convert a name into initials. This kata strictly takes two words with one space in between them.
    </h4>
    <div
      className="card-buttons-container"
    >
      <button
        className="save-to-storage-btn"
        onClick={[Function]}
      >
        Mark as Completed!
      </button>
      <button
        className="see-solution-btn"
        onClick={[Function]}
      >
        See Solution
      </button>
    </div>
  </article>
</div>
`;
